Add documentation for filterComponentOptions (#77)
infinum/eightshift-docs: website/docs/basics/helpers-javascript.md

**Usage:**

```js
	import { dynamicImport } from '@eightshift/frontend-libs/scripts/helpers';

	dynamicImport(require.context('./../../custom', true, /assets\/index.js$/));
```

## elementChildrenHeight

> Returns height of the element measured by height of its children.

Reviewer: ```suggestion
Returns the height of the element measured by the height of its children.
```

**Usage:**

```js
	import { selector } from '@eightshift/frontend-libs/scripts/helpers';

	selector(!(buttonContent && buttonUrl), `${componentClass}-placeholder`);

	selector(test, componentClass, elementClass);

	selector(!newTest, componentClass, elementClass, modifierClass);
```

**Equivalent:**

```js
	!(buttonContent && buttonUrl) ? `${componentClass}-placeholder` : '';

	test ? `${componentClass}__${elementClass}` : '';

	!(newTest) ? `${componentClass}__${elementClass}--${modifierClass}` : '';

```

## getOptionsColor

Use this hook to filter the global colors out of the component or block manifest

* @param {array} colors Array of colors to filter.

**Usage:**

```js
import { getOptions } from '@eightshift/frontend-libs/scripts/editor';

<ColorPaletteCustom
		label={
			<>
				<Icon icon={icons.color} />
				{__('Color', 'eightshift-frontend-libs')}
			</>
		}
		colors={getOptionColors(getOptions(manifest, componentName, 'color', options))}
		value={headingColor}
		onChange={(value) => setAttributes({ [`${componentName}Color`]: value })}
/>
```

## getOptions

> Provides ability to override component options from the parent block/component

Reviewer: ```suggestion
Provides the ability to override component options from the parent block/component
```

> The components must have the same options name as attribute standard with componentName prefix.

Reviewer: ```suggestion
The components must have the same option names as attribute standard with `componentName` prefix.
```

* @param {object} manifest Original manifest from the component.
* @param {string} componentName Current componentName. This is changed depending on the passed componentName.

> * @param {string} attribute Attribute name to check without componentName prefix. Value is auto titleCased.

Reviewer: ```suggestion
* @param {string} attribute Attribute name to check without componentName prefix. Value is auto camelCased.
```

What is `titleCased`?

> * @param {object} options Options provide by the parent block/component.

Reviewer: ```suggestion
* @param {object} options Options provided by the parent block/component.
```

**Usage:**

```js
	import { getPaletteColors } from '@eightshift/frontend-libs/scripts/editor';

	getPaletteColors();
```

## outputCssVariablesGlobal

> Get Global Manifest.json and return globalVariables as css variables.

Reviewer: ```suggestion
Get global manifest.json and return `globalVariables` as CSS variables.
```

			"max":  300,
			"step": 10
		},
		"sectionInSpacing": {
			"min":  0,
			"max":  300,
			"step": 10
		},
		"colors": [
			{
				"name": "Infinum",
				"slug": "infinum",
				"color": "#D8262C"
			},
			{
				"name": "Black",
				"slug": "black",
				"color": "#111111"
			}
		]
	}
};
```

**Usage:**

```js
import { getUnique } from '@eightshift/frontend-libs/scripts/editor';
import globalSettings from './../../manifest.json';

outputCssVariablesGlobal(globalSettings);
```

**Output:**

```js
<style>
	:root {
		--global-max-cols: 12;
		--global-breakpoints-mobile: 479;
		--global-breakpoints-tablet: 1279;
		--global-breakpoints-desktop: 1919;
		--global-breakpoints-large: 1920;
		--global-containers-default: 1330px;
		--global-gutters-none: 0;
		--global-gutters-default: 25px;
		--global-gutters-big: 50px;
		--global-section-spacing-min: -300;
		--global-section-spacing-max: 300;
		--global-section-spacing-step: 10;
		--global-section-in-spacing-min: 0;
		--global-section-in-spacing-max: 300;
		--global-section-in-spacing-step: 10;
		--global-colors-infinum: #D8262C;
		--global-colors-black: #111111;
		--global-colors-white: #FFFFFF;
	}
</style>
```

## outputCssVariables

> Get component/block options and process them in css variables.

Reviewer: ```suggestion
Get component/block options and process them in CSS variables.
```

> If the attribute key has `variable` key than that attribute will be outputted to the inline style for css variables.
> 
> If you provide an additional key `color`, the helper will check global manifest and output css variable for color name.

Reviewer: ```suggestion
If the attribute key has a `variable` key then that attribute will be outputted to the inline style for CSS variables.

If you provide an additional key `color`, the helper will check the global manifest and output the CSS variable for the color name.
```
